src/math/exp_fcn.c

```diff
@@ -0,0 +1,212 @@
+ * Add them all up, taking carry into consideration.
+ *
+ * The 64 x 64 bit multiplication of operands in_0 and in_1 is
+ * shown in the image below. The 64-bit operand in_0, in_1 is
+ * represented by the notation in0_H, in1_H for the top 32 bits
+ * and in0_L, in1_L for the bottom 32 bits.
+ *
+ *				in0_H : in0_L
+ *			x	in1_H : in1_L
+ *			---------------------
+ *	P0			in0_L x in1_L
+ *	P1		in0_H x in1_L		64 bit inner multiplication
+ *	P2		in0_L x in1_H		64 bit inner multiplication
+ *	P3	in0_H x in1_H
+ *			--------------------
+ *			[64 x 64 bit multiplication] sum of inner products
+ * All combinations are multiplied by one another and then added.
+ * Each inner product is moved into its proper power location, given the names
+ * of the inner products, redoing the addition where 000 represents 32 zero
+ * bits. The inner products can be added together in 64 bit addition. The sum
+ * of two 64-bit numbers yields a 65-bit output.
+ *		   (P0H:P0L)
+ *		P1H(P1L:000)
+ *		P2H(P2L:000)
+ *	P3H:P3L(000:000)
+ *	.......(aaa:P0L)
+ * By combining P0H:P0L and P1L:000. This can lead to a carry, denote as CRY0.
+ * The partial result is then multiplied by P2L:000.
+ * We call it CRY1 because it has the potential to carry again.
+ *	(CRY0 + CRY1)P0H:P0L
+ *	(	 P1H)P1L:000
+ *	(	 P2H)P2L:000
+ *	(P3H:	 P3L)000:000
+ *	--------------------
+ *	(ccc:bbb)aaa:P0L
+ * P1H, P2H, and P3H:P3L are added to the carry CRY0 + CRY1. This increase will
+ * not result in an overflow.
+ *
+ */
+static inline void mul_s64(int64_t in_0, int64_t in_1, uint64_t *ptroutbitshi,
+				uint64_t *ptroutbitslo)
+{
+	uint64_t absin0 = ABS(in_0);
+	uint64_t absin1 = ABS(in_1);
+	uint64_t in0hi = absin0 >> 32;
+	uint64_t in0lo = absin0 & UINT32_MAX;
+	uint64_t in1hi = absin1 >> 32;
+	uint64_t prodhilo;
+	uint64_t prodlohi;
+
+	absin0 = absin1 & UINT32_MAX;
+	/* multiplication */
+	prodhilo = in0hi * absin0;
+	prodlohi = in0lo * in1hi;
+	absin0 *= in0lo;
+
+	in0lo = absin0 + (prodlohi << 32);
+	absin1 = in0lo < absin0 ? 1 : 0;
+	absin0 = in0lo;
+	/* shift and add */
+	in0lo += prodhilo << 32;
+	if (in0lo < absin0)
+		absin1++;
+	/* carry */
+	absin0 = absin1 + in0hi * in1hi + (prodlohi >> 32) + (prodhilo >> 32);
+	/* 2's complement */
+	if (in_0  && in_1  && (in_0 > 0) != (in_1 > 0)) {
+		/* bit inversion */
+		absin0 = ~absin0;
+		in0lo = ~in0lo;
+		/* add to low byte */
+		in0lo++;
+		if (!in0lo)
+			absin0++;
+	}
+	/* pointer- output high and low bytes */
+	*ptroutbitshi = absin0;
+	*ptroutbitslo = in0lo;
+}
+
+/*
+ * Arguments	: int64_t a
+ *		  int64_t b
+ * Return Type	: int64_t
+ */
+static inline int64_t lomul_s64_sr_sat_near(int64_t a, int64_t b)
+{
+	uint64_t u64_rhi;
+	uint64_t u64_rlo;
+
+	mul_s64(a, b, &u64_rhi, &u64_rlo);
+	const bool roundup = (u64_rlo & BIT_MASK_LOW_Q27P5) != 0;
+
+	u64_rlo = (u64_rhi << 36 | u64_rlo >> 28) + (roundup ? 1 : 0);
+	return u64_rlo;
+}
+
+/* f(x) = a^x, x is variable and a is base
```

Comment:
Where is / what is `a`? 
```suggestion
/* f(x) = e^x
```

Reply:
The constant `e` has the value 2.718281828459045.
where `a` is a number with a range of values. so, it is not necessary to alter the current depiction.